backends-common/elasticsearch/src/test/java/org/apache/james/backends/es/ElasticSearchHealthCheckTest.java

```diff
@@ -0,0 +1,122 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.backends.es;
+
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.core.healthcheck.Result;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.client.AdminClient;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.ClusterAdminClient;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlocks;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.cluster.routing.RoutingTable;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
```

Comment:
@chibenwa yes I'll add an additional test. I actually started out with your suggestion, but the test now check the cluster state and reports that back. I think I will need to add some additional checks in case elasticsearch is not running or unreachable. I'll see what I can do.

Reply:
The mocked tests you wrote are actually valuable as they allow to check result of inconsistent indexes, what is really hard to recreate in an integration test.

Reply:
I've tried getting this to work, but I seem to have issues with the transport client connecting to the docker container. Any tips? I noticed in JAMES-1952 that I'm not the only one.

Reply:
Mmmh I just succeeded to run these ignored tests in a repeatable fashion.

Regarding testcontainer usage:
 - Did you use a waitStreagy?

Something like :

```
    private static final WaitStrategy WAIT_STRATEGY = Wait.forHttp("/").forPort(ES_HTTP_PORT).withRateLimiter(RateLimiters.DEFAULT);

    private final SwarmGenericContainer es1 = new SwarmGenericContainer(DOCKER_ES_IMAGE)
        .withAffinityToContainer()
        .withExposedPorts(ES_APPLICATIVE_PORT)
        .waitingFor(WAIT_STRATEGY);
```

(This ensure container.start() returns only when ES is up...)

 - Rather than `stop()` `pause()` is to be used to simulate a failure (no answer)

I do not know if it helps...

Reply:
Yes I tried that. What platform are you running your tests on? I'm running on a Mac and I think it has to do with the ip address of the container and something with the transport client. Since docker still runs in its own 'VM' on Mac this might be the reason why it's not working.

Reply:
Hi @jreijn , you can not connect to your docker from a mac machine. From docker for mac website: [Docker for Mac can’t route traffic to containers.](https://docs.docker.com/docker-for-mac/networking/#port-mapping)

What you can do instead, is try to connect your docker on mapped port on host machine. Given declared container above
```java
    private static int ELASTIC_SEARCH_HTTP_PORT = 9200;
    private static int ELASTIC_SEARCH_PORT = 9300;
    private static final WaitStrategy WAIT_STRATEGY = Wait.forHttp("/").forPort(ES_HTTP_PORT).withRateLimiter(RateLimiters.DEFAULT);
    private static SwarmGenericContainer elasticSearchContainer = new SwarmGenericContainer(Images.ELASTICSEARCH)
        .withExposedPorts(ELASTIC_SEARCH_HTTP_PORT, ELASTIC_SEARCH_PORT).waitingFor(WAIT_STRATEGY);
    
    private Client client;
    
    @BeforeEach
    void setup() {
        client = ClientProviderImpl.forHost("localhost", elasticSearchContainer.getMappedPort(9300))
            .get();
    }
```

and don't forget to start and stop your docker before and after all tests
```java
    @BeforeAll
    static void startContainer() {
        elasticSearchContainer.start();
    }

    @AfterAll
    static void stopContainer() {
        elasticSearchContainer.stop();
    }
```

That is the workaround to fix this problem on mac. But we prefer to use containers's IPs and ports in our tests to make it look closer to the real environment. Right after your test are succeed, you could update `"localhost"` to `elasticSearchContainer.getHostIp()` and `elasticSearchContainer.getMappedPort(ELASTIC_SEARCH_HTTP_PORT)` to `ELASTIC_SEARCH_HTTP_PORT`, and let CI verify the tests on linux environment for you!